unit_tests/test_redis_client_base.py

```diff
@@ -22,114 +19,62 @@ def redis_client_auto_register():
+               f"{area_id}/response/unregister_participant": redis_client_auto_register._on_unregister,
+               f"{area_id}/*": redis_client_auto_register._on_event_or_response,
+               "aggregator_response": redis_client_auto_register._aggregator_response_callback})
+
+    def test_function_aggregator_response_callback(self, redis_client_auto_register):
+        message = {'data': {"transaction_id": f"{transactionID}"}}
+        message["data"] = json.dumps(message['data'])
+        redis_client = redis_client_auto_register
+        redis_client._transaction_id_buffer.append(transactionID)
+        redis_client._aggregator_response_callback(message)
+        assert redis_client._transaction_id_buffer == []
+
+    def test_function_check_buffer_message_matching_command(self, redis_client_auto_register):
+        message = dict()
+        message["data"] = {"transaction_id": f"{transactionID}"}
+        redis_client_auto_register._blocking_command_responses = {"register": {"transaction_id": f"{transactionID}"}}
+        assert redis_client_auto_register._check_buffer_message_matching_command_and_id(message['data']) is None
+
+    def test_check_buffer_message_matching_command_and_id_throws_exception(self, redis_client_auto_register):
         message = {}
         with pytest.raises(RedisAPIException,
                            match='The answer message does not contain a valid '
                                  '\'transaction_id\' member.') as ex:
-            redis_client_base = RedisClientBase(area_id=area_id, autoregister=False)
-            redis_client_base._check_buffer_message_matching_command_and_id(message)
+            redis_client = redis_client_auto_register
+            redis_client._check_buffer_message_matching_command_and_id(message)
 
-    def test_register_blocking_true_throws_exception(self):
+    def test_check_buffer_message_matching_command_and_id_throws_another_exception(self, redis_client_auto_register):
+        message = dict()
+        message["data"] = {"transaction_id": "1234"}
+        with pytest.raises(RedisAPIException,
+                           match='There is no matching command response in _blocking_command_responses.') as ex:
+            redis_client = redis_client_auto_register
+            redis_client_auto_register._blocking_command_responses = {
+                "register": {"transaction_id": f"{transactionID}"}}
+            redis_client._check_buffer_message_matching_command_and_id(message['data'])
+
+    def test_function_register(self, redis_client_auto_register):
+        data = {"name": area_id, "transaction_id": area_id}
+        redis_client = redis_client_auto_register
+        redis_client.redis_db.publish = MagicMock()
+        redis_client.register(is_blocking=False)
+        redis_client.redis_db.publish.assert_called_once_with(f'{area_id}/register_participant', json.dumps(data))
```

Comment:
What are you testing here? 
This is the same as https://github.com/gridsingularity/d3a-api-client/pull/163#discussion_r691428221

Reply:
I am testing whether the redis_db. publish method is called with correct arguments.

Reply:
Sorry, it was too late yesterday. You are right. ignore me.